lib/datapacksjob.js

```diff
@@ -3016,10 +3020,71 @@ DataPacksJob.prototype.downloadPerformanceData = async function (jobInfo) {
                 records.push(record);
             })
             .on("end", async function () {
-                var res = records.map(item => {
-                    return item[field_name];
+               
+                var omni_id = records.map(item => {
+                    return JSON.parse(item[field_name]).OmniScriptId;
                 });
 
+                var unique_omni_id = Array.from([...new Set(omni_id)]);
+
+                var res = [];
+
+                while (unique_omni_id.length > 0) {
+
+                    var chunk = unique_omni_id.splice(0, 150);
+                    var unique_omni_id_format = "";
+
+                    for (var j = 0; j < chunk.length; j++) {
+                        if (j == (chunk.length - 1)) {
+                            unique_omni_id_format = unique_omni_id_format + `'${chunk[j]}'`
+                        } else {
+                            unique_omni_id_format = unique_omni_id_format + `'${chunk[j]}',`;
+                        }
+                    }
+
+                    var additional_data = await self.vlocity.jsForceConnection.query(`SELECT ${self.vlocity.namespace}__PropertySet__c,Name,${self.vlocity.namespace}__OmniScriptId__c FROM ${self.vlocity.namespace}__Element__c WHERE ${self.vlocity.namespace}__OmniScriptId__c IN (${unique_omni_id_format})`);
+
+                    additional_data.records.map(item => {
+
+                        var name = item['Name']
+                        var omniscript_id = item[self.vlocity.namespace + '__OmniScriptId__c'];
+                        var propertyset = JSON.parse(item[self.vlocity.namespace + '__PropertySet__c']);
+
+                        records.map(item => {
+                            var element_data = JSON.parse(item[field_name])
+                            var element_name = element_data.ElementName;
+                            var element_omniScript_id = element_data.OmniScriptId;
+                            var element_type = element_data.ElementType;
+
+
+                            if ((element_name == name) && (element_omniScript_id == omniscript_id)) {
+                                switch (element_type) {
+                                    case "Calculation Procedure Action":
+                                        element_data.configurationName = propertyset.remoteOptions.configurationName;
+                                        break;
+                                    case "DataRaptor":
```

Comment:
This should be:
case "DataRaptor Extract Action":
case "DataRaptor Post Action":
case "DataRaptor Transform Action":
case "DataRaptor Turbo Action":